### ISSUE TITLE
example/mnist/mnist_conv.rs run failed 

### ISSUE THREAD

**Issue by ChenHuaYou** [LaurentMazare/tch-rs#434]
if i run the example in main branch i would failed.    
 Running `target/debug/examples/mnist conv`
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Torch("one of the variables needed for gradient computation has been modified by an inplace operation: [CUDAFloatType [256, 1024]], which is output 0 of ReluBackward0, is at version 1; expected version 0 instead. Hint: enable anomaly detection to find the operation that failed to compute its gradient, with torch.autograd.set_detect_anomaly(True).\nException raised from unpack at ../torch/csrc/autograd/saved_variable.cpp:164 (most recent call first):\nframe #0: c10::Error::Error(c10::SourceLocation, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >) + 0x6b (0x7f56411f00db in /home/mc/mydata/data/libtorch/lib/libc10.so)\nframe #1: c10::detail::torchCheckFail(char const*, char const*, unsigned int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) + 0xce (0x7f56411ebd2e in /home/mc/mydata/data/libtorch/lib/libc10.so)\nframe #2: torch::autograd::SavedVariable::unpack(std::shared_ptr<torch::autograd::Node>) const + 0x914 (0x7f5644f9bfe4 in /home/mc/mydata/data/libtorch/lib/libtorch_cpu.so)\nframe #3: torch::autograd::generated::ReluBackward0::apply(std::vector<at::Tensor, std::allocator<at::Tensor> >&&) + 0xdb (0x7f56447f896b in /home/mc/mydata/data/libtorch/lib/libtorch_cpu.so)\nframe #4: <unknown function> + 0x3b2ebdd (0x7f5644f5ebdd in /home/mc/mydata/data/libtorch/lib/libtorch_cpu.so)\nframe #5: torch::autograd::Engine::evaluate_function(std::shared_ptr<torch::autograd::GraphTask>&, torch::autograd::Node*, torch::autograd::InputBuffer&, std::shared_ptr<torch::autograd::ReadyQueue> const&) + 0x1517 (0x7f5644f59d17 in /home/mc/mydata/data/libtorch/lib/libtorch_cpu.so)\nframe #6: torch::autograd::Engine::thread_main(std::shared_ptr<torch::autograd::GraphTask> const&) + 0x5a6 (0x7f5644f5a9f6 in /home/mc/mydata/data/libtorch/lib/libtorch_cpu.so)\nframe #7: torch::autograd::Engine::thread_init(int, std::shared_ptr<torch::autograd::ReadyQueue> const&, bool) + 0x99 (0x7f5644f51ff9 in /home/mc/mydata/data/libtorch/lib/libtorch_cpu.so)\nframe #8: <unknown function> + 0xd6de4 (0x7f56a095dde4 in /lib/x86_64-linux-gnu/libstdc++.so.6)\nframe #9: <unknown function> + 0x9609 (0x7f564113f609 in /lib/x86_64-linux-gnu/libpthread.so.0)\nframe #10: clone + 0x43 (0x7f5640f11293 in /lib/x86_64-linux-gnu/libc.so.6)\n")', src/wrappers/tensor.rs:322:27

but if i run in pytorch-1.8 and libtorch1.8 i would okay. why?

**Comment by ChenHuaYou** [comment]
31 impl nn::ModuleT for Net {
   32     fn forward_t(&self, xs: &Tensor, train: bool) -> Tensor {
   33     ¦   ¦   let x = xs.view([-1, 1, 28, 28]); ‣x: Tensor
   34     ¦   ¦   let x = x.apply(&self.conv1); ‣x: Tensor
   35     ¦   ¦   let x = x.max_pool2d_default(2); ‣x: Tensor
   36     ¦   ¦   let x = x.apply(&self.conv2); ‣x: Tensor
   37     ¦   ¦   let x = x.max_pool2d_default(2); ‣x: Tensor
   38     ¦   ¦   let x = x.view([-1, 1024]); ‣x: Tensor
   39     ¦   ¦   let x = x.apply(&self.fc1); ‣x: Tensor
   40     ¦   ¦   let x = x.relu(); ‣x: Tensor
   41     ¦   ¦   let x = x.dropout(0.5, train); ‣x: Tensor
   42     ¦   ¦   let x = x.apply(&self.fc2); ‣x: Tensor
   43     ¦   ¦   x
   44     }
   45 }

i finally find the answer that in the main branch , we should use dropout instead of dropout_ if we wan to backward.
